src/debug/flutter_debug_impl.ts

```diff
@@ -250,6 +250,10 @@ export class FlutterDebugSession extends DartDebugSession {
 		if (args.args)
 			allArgs = allArgs.concat(args.args);
 
+		if (!!args.daemonPort) {
+			allArgs = allArgs.concat(`--daemon-connection-port=${args.daemonPort}`);
+		}
```

Comment:
This code here is the legacy DAP, so this change will be lost in the mode to the SDK DAPs.

While we could also duplicate it in the SDK, I think a better way would be to insert it into `toolArgs` in the VS Code extension, which both DAPs accept as a generic way to pass additional flags without specific support in the debug adapter.

See `buildFlutterToolArgs` here:

https://github.com/Dart-Code/Dart-Code/blob/3047a27cbfde9844d1a1bbe20a40780af7e37c40/src/extension/providers/debug_config_provider.ts#L626

This is the code that adds things like `--web-renderer` based on settings. There is also a `buildFlutterTestToolArgs` if you need to include it for test.

(if this flag only works for the internal Flutter script, we should probably also ensure we only inject it if that script is set, so if an internal user opens a standalone project we don't send it to the public daemon and fail).

Reply:
Ah okay, I forgot to test with SDK DAPs. I added a check based on `customTool` and `daemonPort`.

I think we won't need this for tests because tests use the "device" that's on the same machine as our extension.

Reply:
I think we can (should?) remove this code now, since it's being handled with `toolArgs`? (that way we won't be surprised with any differences between SDK/Dart-Code DAPs).